pkg/apis/apps/validation/validation.go

```diff
@@ -75,6 +75,48 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("podManagementPolicy"), spec.PodManagementPolicy, fmt.Sprintf("must be '%s' or '%s'", apps.OrderedReadyPodManagement, apps.ParallelPodManagement)))
 	}
 
+	switch spec.UpdateStrategy.Type {
+	case "":
+		allErrs = append(allErrs, field.Required(fldPath.Child("updateStrategy"), ""))
+	case apps.OnDeleteStatefulSetStrategyType, apps.RollingUpdateStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition != nil {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					fldPath.Child("updateStrategy").Child("partition"),
+					spec.UpdateStrategy.Partition.Ordinal,
+					fmt.Sprintf("only allowed for updateStrategy '%s'", apps.PartitionStatefulSetStrategyType)))
+		}
+	case apps.PartitionStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition == nil {
```

Comment:
> What I like about this approach is that operators, human or otherwise, have an opportunity to validate the deployed product prior to triggering the controller to signal completion. Readiness checks are good for determining if the product is running, but this would allow for adding a smoke test prior completing the update.

Not sure why this is different in the case partitioning is an option of the rolling strategy. A smoke test prior to completing the update, or sending out an e-mail, or any kind of post-hook action could be possible to trigger by setting the ordinal to the size of the statefulset (sts.spec.replicas). At that point, the controller has moved all replicas to use the updated revision but it still waits for a higher-level orchestrator to set the ordinal to nil in order for the update to be considered as complete.

Another reason for moving this as an option vs a separate strategy is that in the future adding new strategies will make validation and UX even more complex and you will likely need to create an additional partition strategy for the new strategy. Take for example Deployments, we support Rolling and Recreate. Adding a single Partitioned strategy is not going to work for both, adding a PartitionedRolling and PartitionedRecreate will be a UX hit.